### ISSUE TITLE
Planing for Dean's quest line.

### ISSUE THREAD

**Issue by Novae17** [taberenc/AOG_2022#1]
Hey chief, I've already got all I need to start modifying the game.
Got all the tools ready and already put some work into it.
I'm currently finishing the house, (I used the one on sale at the top left of the map just beside the orphanage.)
As soon as I got it ready, I will create a way for Garnet to buy it.

**Comment by taberenc** [comment]
Great! If you have any trouble with the animation of the door, or some switch or common even.. just tell me. I'm not an expert, but i know a bit how to manage these kind of issues in this game. In this case i guess that you will work with variables or conditionals.
There are also maps to do in charthage and room for npc's and events. In ikary and clairia also.
Wellcome!

**Comment by Novae17** [comment]
I managed to get doors working for the most part, I'm currently having issues with them when I need to enter another room inside the same house. It's nothing I won't be able to fix though.

I do have a background in IT, so I'm sure I can find a way to fix any issues I encounter. I'll get the house done, then try for a pull request and see if it works out! 

Also, I want to try and focus on a single quest at a time, finishing them one by one... I'm not sure how much content I want to give to Dean's questline but I was planning to give him substantial content. I guess we will see...
Slow and steady wins the race!

**Comment by Novae17** [comment]
Got the house and Basement done with working triggers and everything.
Will start to work on a way to buy the house now.

Once the map is all done and the process of buying it is completed, I'll do a pull request.

**Comment by taberenc** [comment]
the headache is that, since for the last months, i've been working alone on this, the debug shortcuts were active. Then, when i had already forgot about it, the game has had its own f95 thread.. .with those shortcuts , which can make the game to contain errors for someone who plays it as 1st time.
And, while i did as soon as i was aware of the problem, an update with that mistake fixed, the update it seems to take ages to be done in the thread 1st post.
There were also a pair of bugs: in the respawning on guild adventurer's bed and one related to shame in the early game.
This kind of bugs are the things that i wanted to be detected before a release.. because people it seems to be very exigent, and the game is not finished.
the impact of the f95 forum also it seems to be replicated in other websites that take material from this forum for their websites.. and i hardly doubt that they will update. so .. this is all like a nightmare come true.
And most of users don't seem prone to help to fix those bugs, just to complain.
If i would have know how this forum worked beforehand.. and that i wouldnt be able to do the updates on the thread..!

Patience, and not losing the focus about the goal U_U

**Comment by taberenc** [comment]
do the request when you think that is correct and i'll add it and try it.

**Comment by Novae17** [comment]
Yeah, I've been lurking for ages on Forums like that, only now picking up the work.
I know exactly what you mean, but we can't let that stop us in any way. 

People will be assholes and that's that. xD

I should be done tonight, I put in 2 hours a night on it, nothing more, nothing less...
Discipline > Passion. Once I'm done with that house, I could try to see if I can fix that issue.
It might be as simple as it's not giving HP to the MC, which calls a game over. (The bug with the guild respawn.)

**Comment by Novae17** [comment]
I guess what I mean to say is that, something calls the game over status, we just need to find it~

**Comment by Novae17** [comment]
About to do a pull with the house... should not take too long~
I've synced your main first, downloaded it, and made it work in your main branch.
So everything should be working exactly as it was when you uploaded it~

**Comment by Novae17** [comment]
Sent in the request, hope everything works out!

**Comment by taberenc** [comment]
Hi.. i'll check it out.
The bug about being knocked out, waking up on the adventurer's guild and then dying again has been fixed, as far as i know.
The debug stuff has been removed in the most part. The one that was already on the game is still there. If you want to activate, is in the upper side of the table in the mid of the house of garnet. There you obtain the debug item and several npc's get active to give you items and stuff. that's the debug that was provided years ago and i leave it there.
Another bug that has happened was in the main portal of wall castle. Castle gets defenders (and the door) after switch heladorc06, which is activated by the abbot in the ceremony that let garnet to have access to the hell area.
before that, castle is empty (there is just a guy there and a goat).

by now there are not more bugs.. beyond the ones caused by the unaligment in switches and access to areas that the debug tools cause. The time that the dudes in the web take to update the files is something that i can't fix.

so..! about things to do.. yea, there are a lot. I'll provide you a list later, if you want. but meanwhile..  one thing pending to do is the scene in the 2nd floor of the library of Dali, when Garnet learns magic. Is supposed to be a sex scene with tentacles.
I mention it because is in the early game.
Other things easier to do are on the towns. The other day a guy asked in the forum about scenes in clairia, iirc. You have plenty of room there for indoors, characters.. 
Dali is a bit packed already, but has a few houses still empty of npc's or events. In the main town, at least 2 houses are not developed.
In the neighbourhood in the left side (where the abandoned house), i think that the inside of the houses is all done, but 2 or 3 of them doesnt have still any npc nor event. check them out, if you feel in the mood.
also, north of the city, in the forest, .. that area is intended for very early game. There are no enemies there nor anything. Is just ot know about the area where is the purple rock. The point is.. maybe some soft scene (like a masturbation in the wild, without rape nor anything like that) could be added there. don't know.
by the way, to make npc's, i the hell area i did ones without legs, that are supposed to be spirits waiting. I did them with photoshop and a program called "character maker xp". i  would recommend to try it and know a bit about it. it can be very useful.

**Comment by Novae17** [comment]
I'll look up the "character maker xp" program for sure.
I was looking forwards to fixing some bugs but it's good to know most of them have been fixed.

I'll continue with Dean's side quest, then I'll pick up another one you listed after letting you know~
Oh and, for sex scenes, I have around 3 years of ERP experience on F-list with all sorts of kinks, so I should be able to cook up some nice scenes, with a bit of effort.

Let me know how the merge worked out~

**Comment by taberenc** [comment]
i've visited the house. Also seen the map.
About the doors.. they just transfer the locations, but dont move. The curtain, as example, or the metal door. You mind if i change it and make them move?
also.. the bottom part of the map of the 1st room.. is full of events in autorun set, but .. for what they are there? its purpose is to make as a path blocking? because that wall is already path blocking. they are not needed if thats the purpose. yes, they partially hidden Garnet.. but this belong to a 3d effect to give the sensation that the wall is betwen the player and garnet. I think that is not necessarily bad.
what do you think?

**Comment by taberenc** [comment]
I've modified the door beside the kennel. map352. Check it out.

**Comment by taberenc** [comment]
by the way, if you want to see the changes, download the new map351, map352 and the png file: 172-door3-b.png
check the changelog too, to see if you agree or want to add something there. You know better than me the events related to these maps.

**Comment by Novae17** [comment]
I see what you did, nice!
I had issues figuring out how to fix them due to some strange behaviors.
I'll take a look at how you did it and replicate it next time.
I will also fix that wooden door tonight.
I'm going to fix the main door issue while changing maps.

now for the triggers that block paths, Yeah... 
I had to find a way to block it to avoid the MC from passing under the walls.
I will try to find a more... discrete way of fixing it.

Also, please... feel free to fix or change anything I do. 
I'm sure the final product will be better if we both work to make it the best possible!
I'm not attached to anything I work on that much... 
I know some people would be annoyed at it, but I see it as a net positive.

I've synced my branch with the main one and will get on all of this later today~

**Comment by Novae17** [comment]
![image](https://user-images.githubusercontent.com/130022673/230767386-4e4ce5d2-6a0c-4474-9f44-512eaff88169.png)
Found a fix with the curtains, and that might be needed for more doors.
To avoid an issue where the doors glitch out when you open them from the top, you need to check both:
'move animation' And 'Direction Fix'

**Comment by taberenc** [comment]
yea.. it could be. I did that was late night, and it was not even my intention, just to check the map.
About the blocks of the wall.. try it without them. Garnet partially gets behind the wall, but thats fine. is what gives 3d sensation. i think that she wont be able to lave the room. i may be wrong though. so.. try it, and let see.

I've seen that there are several scenes in the bed.. can you explain the story behind them? are copy-pasted from other places or you made them? whats the purpose in character's story arch? i mean.. is an achievement after some mission or something?

**Comment by Novae17** [comment]
I left the bed scenes there to give me an idea of how to build them. They should be harmless for now but I will need to remove them once I get at least one working in that room. (It's mostly just a placeholder.)

Dean is the thug in the orphanage, the one that places a collar on the desk upstairs and asks Garnet to put it on.
I'm planning to give him a small story arc where he slowly turns Garnet into a mindless pet, trapping her inside the house for a potential game-over. I have not made any sex scenes yet, I wanted to get the house done and finished so I could focus on working with a finished environment first.

Regarding Photoshop, I usually use Gimp, but both do the same thing. I will need to make a custom png for that door if I want to use it. Also, yes... I've had to change the color of that blue mailbox outside the house as it's normally red.

Now, for the wall thing, the issue is that you can slip under the wall and bed in the bedroom. I will remove the events and place some terrain that you can't walk on and simply place the wall as an upper layer to fix the issue. I wish I had thought of it before doing the event thing... but it should not cause too many issues for now. After that, I really need to get at least one scene working, since players are going to be mad if they farm 20k gold for the house and there is nothing inside of it.

**Comment by taberenc** [comment]
As a starter, and since is her house, you could begin with masturbation. just an idea.
It would be easy to do: it happens in several scenes of the game already. the kind of masturbation would be up to you.
btw.. the house is accessible right now without buying it, no? just asking.
the house itself is great. i like it.

**Comment by Novae17** [comment]
The house is not accessible unless you buy it, It's 20k.
The only way for Garnet to want to buy it is if she is Dean's slave in particular. The variable: 0026 set at 3
![image](https://user-images.githubusercontent.com/130022673/230771996-cafc3733-ec31-4a80-a1d6-588396d3cf8a.png)

I can't really do a masturbation scene since Dean asked that she gets a chastity belt and give him the key.
I will see how to make that work too...

**Comment by Novae17** [comment]
I've run into an issue, I need to find a way to make the object Dean's collar work while being a slave or a dog... 
I think I'm going to change it into a simple metal tag and turn the item into an accessory that Garnet can wear at the same time as the collar. That way I don't have to touch any other triggers, safe for the one with said tag.

The last thing I want is to start to play with the Adventurer/Slave/Dog status.

**Comment by Novae17** [comment]
Making a PNG for normal doors.
![image](https://user-images.githubusercontent.com/130022673/230784017-d9ac7e09-227f-4263-aeb3-33ce81e99593.png)

**Comment by taberenc** [comment]
> The house is not accessible unless you buy it, It's 20k. The only way for Garnet to want to buy it is if she is Dean's slave in particular. The variable: 0026 set at 3 ![image](https://user-images.githubusercontent.com/130022673/230771996-cafc3733-ec31-4a80-a1d6-588396d3cf8a.png)
> 
> I can't really do a masturbation scene since Dean asked that she gets a chastity belt and give him the key. I will see how to make that work too...

no no, if masturbation does not have sense in this context, forget about masturbation. was just an idea. coherence is more important.

**Comment by taberenc** [comment]
> I've run into an issue, I need to find a way to make the object Dean's collar work while being a slave or a dog... I think I'm going to change it into a simple metal tag and turn the item into an accessory that Garnet can wear at the same time as the collar. That way I don't have to touch any other triggers, safe for the one with said tag.
> 
> The last thing I want is to start to play with the Adventurer/Slave/Dog status.
> 
> Would you happen to know which script I need to work with to add the slave tag to her sprite?

be a slave and at the same time be a dog? mmm.. the complication that i see is.. that this would mean like make a different kind of collar dog .. and that would mean to remake all the interactions of garnet with npc's that reaction to dog status, and add another interaction with the new situation. It could be done, but would be a lot of job.
i've just come back home now. I need to dinner. i'll be back later and think about this.

**Comment by taberenc** [comment]
the iron dog collar would be just like a new necklace. would have its properties (def, atk, and so on), be or not be removable (i guess that it wouldn't) and could be in the collar itself that certain attributes could be activated (like hypnosis effect or whatever you chose). Then.. if is not a bad end, would be needed to think how or where that collar is removed. Not if is a bad end.

**Comment by taberenc** [comment]
ok, sure. it may be a way to do what you have in mind. is just a matter of find it.

**Comment by Novae17** [comment]
I will go on a quest for the holy grail.
If I can find where they refer to clothes, we're going to be able to clothing items and weapons.

**Comment by taberenc** [comment]
what do you mean? within the script? i already created clothes there. The freelance was the last one. but i think that is too.. naughty, even knowing that freelance is an euphemism for whore.

**Comment by Novae17** [comment]
Oh, I would mostly just need to know which script handles that. (The name of it.)
I will take the time to read it and figure out how it works.

That would save me quite a bit of time~

**Comment by Novae17** [comment]
Oh, by the way...
I do have a 40h job so I won't be able to push in that 2 hours every day. 
I'll still do my best though!

**Comment by taberenc** [comment]
is armor stuff.  there are several relations that are done by different parts of the script. but the basics i think that can be done within the rpg maker itself: create armor, edit  basic characteristics and so on. changing the code is needed when is affected the graphic area, and thats on 3 different leveles (at least): the sprite while playing, the figure standing up on the right side and, in armors, the different stages (new, broken...) and positions that will have depending on which Garnet are we talking about. As example, in a collar, when she is in shame, the neck changes its position, so the collar needs to do it too.
But if is just about creating an armor, there is no need to look inside the code. You can create it directly using the menus provided by the program, iirc.

btw.. 40h/week is a normal job ( 8hs, 5days). I do also have a job. So no pressure and no hurry. take it easy and do what you can. this is not about speed, is about be constant. So dont worry about that.

**Comment by Novae17** [comment]
Alright, I figured out how the imaging system works. I'll be able to make some new pieces of equipment when needed!
(Standing pictures and all.)
I've progressed a bit on Dean's side quest... I should have my first H-scene done this weekend.

**Comment by taberenc** [comment]
I didn't understand exactly what did you needed. To make an armor.. is within the rpgmaker.
The graphic part.. is when you have done the armor within the rpgmaker, and linked to it. Then you can add it on the garnet figure on the right side of the screen. Is a png file. I did a lot of them. If you need it, i can do it myself.
And then.. there is the sprite, that.. if is the case of the collar.. i think that the existing sprite could work, maybe. Anyways, in the sprite, the collar is very small.

**Comment by Novae17** [comment]
Oh, I simply did not understand how the game pulled the data from the folders.
I've found the script that manages the standing picture and read it a bit to find out that it simply pulls items of the same name in the images.

TLDR, I had not found the 'Persona' script yet and did not understand exactly how it worked. 
I figured it all out now, so I should be good!
I will let you know if I end up needing some more help~

**Comment by taberenc** [comment]
the persona is related to the character as a whole. but you dont need to change that. If you have done an armor, you only need to change that.
The drawing of the armor is only that object. and is overlapped on the drawing of the character (the persona).
You need, first, to create the armor (in the place that i told you, within rpg maker). then, you create the armor drawing (photoshop, as example). The file is a png with transparency that let the player to see the character drawing and, over it, you draw.

I know that is not easy to figure out some things. It took me some time too. And im not fluent english speaker nor i'm even a programmer. Moreover..  sometimes i forget how i did things before.
but i did a lot of armors, so i know this. you make the name, description and characteristics. And then the png file with that same name. when you use that object, the game it will check if the png file that fits with its characteristics is there.
For the sprite the procedure is different, but for the persona stuff is like that. and you dont need to change anything on the pesona thing nor on the code.
My recommendation is that you see how it works in any given armor already existing, like .. i dont know.. the cotton robe, as example. is about the png files and their names.

**Comment by Novae17** [comment]
That's exactly what I did~
I managed to make a placeholder, just something to test it out. and it worked.
So I'm good with making armor and other accessories now!

**Comment by taberenc** [comment]
the only thing that i don't remember right now is where the sprite was changed. i mean.. i know how is done, but it took me some time to figure out how to change it in the last "armor" that i did, the "freelancer". But if i would need to make a new one, i would need to look for it again. 
Btw, freelance is an euphemism of whore. And i wanted to make the "armor" to be evident in that sense, but since i did it that i think that is too evident. i mean.. she is basically nude. But the drawing was not that bad .. so.. im not gone into change it by now. Is a kind of armor that is obtained in the late game, and using it is optional (in case someone could feel offended by how revealing it is), but anyway.. What do you think? do i change it? I think i should, tbh.

**Comment by Novae17** [comment]
You could easily turn it into a dancer's armor and create a new one for the freelancer.
It should be simple enough, I think.

I feel that the only thing wrong with it is the fact that her breasts are exposed.
Maybe you could use the bra from the 'Kai Haiyuan' armor.

**Comment by taberenc** [comment]
i did the drawings in photoshop. is a bit annoying because you should make them for its several states, but once is done, since i've got it there in layers, is not that hard to modify. Just annoying: make the changes and then rename the files and so on. The bulk of the job is already done.
I tend to agree in that it shows too much in its normal state.
but there are some arrmors that still don't have some parts (like the felt armor, which still don't have its own sprite, iirc), so i'll put it on the list of things to do. The felt armor, along as a few more, was in the armor list, but didn't had any graphic representation, as it happened with all the helmets. i did those graphics around december.. or january.

**Comment by Novae17** [comment]
I do have some limited artistic skills.
I'm also able to work with reference pictures to try and give the armors a decent look, especially since the art style is not too fancy.

I'm currently working on a blowjob scene template, something we will be able to use to create more scenes with more characters. when we feel like it. All why keeping the looks as close to the original as possible, of course.

**Comment by taberenc** [comment]
Im working on the tilesets of the ending maps. Since she goes underground and light comes from the floor, i need walls with the light coming from the bottom, and also to be a bit reddish. 
![tilesets_redLight](https://user-images.githubusercontent.com/2058499/232334706-9bbc3145-6281-4fe3-97fb-86fdd6d3a6bf.png)

Working on tilesets is a bit annoying. But i hope that it worths it.

**Comment by taberenc** [comment]
About the scenes you mention.. I've not seen them, but if you think that they are ok, then i guess that they are. If you need me to check them out or something else, tell me.
I did a few sprites and scenes, and i know that is not an easy job. So.. keep it your mood up! And i wish you a good job!

**Comment by Novae17** [comment]
I'm still working on the scenes, I fixed the issue with the collars and removed all the trash lines in the event.
Took a long time to fix, but at least now I'm working with something that's solid.

I see what you mean by the color of the walls.
Would you mind sending me a PNG of the file or uploading it on the main branch?
I will download it and try something to make it fit for a good look~

**Comment by Novae17** [comment]
And don't worry about my mood~
I'm not letting go of that project for a good long while!
I can't promise anything, of course... we never know what could happen... but I have no plan on letting go for now.
I mostly do this in my free time, so I can't output as much as I would want... especially since everything I do is new to me.

I guess what I mean to say is... As long as we both hold on, we should be good~

**Comment by taberenc** [comment]
I've modified tileset 181
Added map 351, which with the fog effects and smoke, I think that works fine.
From this will split in 2 directions. South, towards the underground of the entrance (connecting with the holes on the ground there). To the right, will go to final boss area and black tower

**Comment by Novae17** [comment]
Is'int 351 Dean's house?
I've had it on my branch for a little while.
![image](https://user-images.githubusercontent.com/130022673/232784226-dca5ed5a-34b1-4fd1-a92e-f6b75d3fcc1b.png)
I was planning to import your main branch first then add all the changes I've worked on once I'll be ready to do so.
So it's no big deal.

**Comment by taberenc** [comment]
Im gona check it out. If it is the case (and it could be, both are new), i'll change mine.

**Comment by taberenc** [comment]
You are right. maps 351 and 352 corresponds to Dean's house and Dean's house underground.
I'll rename my new maps now to different numbers and upload yours on the game again, in the version that i've got.
And sorry. My fault.

**Comment by taberenc** [comment]
when you have an update to those maps, make a pull request and i'll add the updates.

**Comment by taberenc** [comment]
now your maps are on the web with their numbers and they work.
now ill try to add mine.

**Comment by taberenc** [comment]
Ok.. now, in my computer, it seems all fixed now. Your maps and mine maps works.
I'll upload the changes in the game now.
I recommend that you download the current version, change the files on it with your current files with your changes, and then make a pull request. Then we both will be working on the same version.

right now im doing maps for the end of the game. So .. well, is better to keep in mind that me, and maybe you, need to update regularly our computer files with the ones in the web.

**Comment by taberenc** [comment]
I think that everything is fine now and that the problem is already fixed.
sorry again.
As said, i recommend to download the current version in the web, change the files that you need once downloaded, in order to keep the work as you have it. And remember to ask for a pull request as much times as you need it. Then we both will work with the same version.
in your pull request add also the modification to the changelog. there you can write the changes that you have done.

If you download the version now in github, you'll see the dean house maps as the last version that i saw and also the files (the job is still half done) in which ive been working.

I think that everything works fine now.

**Comment by Novae17** [comment]
> I think that everything is fine now and that the problem is already fixed. sorry again. As said, i recommend to download the current version in the web, change the files that you need once downloaded, in order to keep the work as you have it. And remember to ask for a pull request as much times as you need it. Then we both will work with the same version. I need to remember to download more frequently and keep my computer version updated. in your pull request add also the modification to the changelog. there you can write the changes that you have done.


Haha, no issues my friend, no need to be sorry at all! <3 
Once I'm done with my changes, I always download your most recent build and then slowly fixate the changes to it.
That way, I'm sure to be able to avoid any issues with shared files like items or common events.

Since I'm working on things that are isolated to 2-3 maps, it reduces the number of interactions with everything else.
If I began to work with you on finishing the end of the game, Then yes, I would need to update and pull as often as possible.

**Comment by taberenc** [comment]
btw.. right now im working on some characters for the upper floor (lust one) of purgatory

ive got a pattern for guys with a bone, and now i should edit the dresses. is a bit tedious work, but i hope that they will be useful

I did that before. that  kind of npc's.. well, they could work better, but i think that they work good enough
![characterWorking](https://user-images.githubusercontent.com/2058499/233204435-b25e22c6-2326-4182-a7e2-6e4bf920602a.png)

**Comment by Novae17** [comment]
Hey~
Just passing by to let you know I'm still working on it.
I'm not as fast as I figured I would be, but I'm putting as much time as I have into it.

As for the sex scenes around that place... I'm not sure what would work.
Maybe something centered around magic? 
Maybe have the cutie lifted up in the air to get fucked by that no-legged man?

**Comment by taberenc** [comment]
in that area there are a bunch of scenes. One is a ghost that possess her and makes her to masturbate. This event is single. but unlocks the possibility to use a green stone that i added on the tileset and use it as much as wanted by the player.
Another one is about peeing (not a fetish of myself, but i know that some people may appreciate that), which is also a way to change from any state (dog , adventurer, slave) to any state. Thats required in the end of the game because magic gets unlocked according to the level, and some of the spells depends on having good level as slave or as dog. So it should be possible, at the end, to have a way to change the state easier and faster.
Another scene is when the comrades of her father are transfered to the living world. that scene involves tentacles.
Finally, there is a scene where 2 tiny demons fuck her mouth and ass at the same time. im thinking on move that to the upper floor (lust).
Those are the scenes that ive done in that hell area by now.
Since the area corresponds to greed, i had planned to add some robbery events.
Then.. in the current level where she is (the greed) i dont think on more scenes, by now, because the upper floor is lust, and if there is a place to add scenes is there. But well, we will see.
The lower floor is wrath (where the final boss is), and maybe will be some scenes there, but just a bunch, if any.

about the time.. no hurry. i took my time too. i mean.. after thinking about it and knowing what i will do, then i do the maps.. modifying tilesets if needed, then put the npc's.. modifying characters if needed, then the dialogues, and is when ive got the maps, characters and main dialogues that i make the events that ive previously planified (not very much, usually). And if i want to keep a good enough quality, i need time.
I know that there is room for more events on maps that ive previously done, but my focus is on the end of the story, and while i add some here and there, i dont want to get too focused on this kind of work (expanding the game), because then i wouldnt never finish it.
Once there is the good end, then, maybe, ill add more scenes here and there, more calmly, maybe longer, and with more attention to details. who knows, maybe i could add some cg's. But its not my priority right now.

**Comment by taberenc** [comment]
about the scenes with ghosts.. it doesnt matter very much if they dont have legs if they float as if they would have them. if they have working dicks on position, then is like if they were any other kind of npc.
I could add some "all the way throught" kind of event, as example.. but in lust floor. A ghost could do that.

**Comment by taberenc** [comment]
same here. we are near the end of semester and job is huge now.

**Comment by taberenc** [comment]
ive undone most of the changes that you did because when ive tested my game has crashed.
I left the deans hose since there is no script involved, i think. the wooden door on the 2nd floor does not move. just teleports.
